src/main/org/nlogo/window/InputBox.scala

```diff
@@ -18,11 +18,19 @@ import javax.swing.text.EditorKit
 import javax.swing.KeyStroke.getKeyStroke
 import javax.swing.{JDialog, JOptionPane, AbstractAction, ScrollPaneConstants, JScrollPane, JButton, JLabel}
 import javax.swing.plaf.basic.BasicButtonUI
+import java.util.prefs.Preferences
 
 abstract class InputBox(textArea:AbstractEditorArea, editDialogTextArea:AbstractEditorArea,
             compiler:CompilerServices, nextComponent:Component)
   extends SingleErrorWidget with Editable with Events.InputBoxLoseFocusEvent.Handler {
 
+  object Prefs {
+    private val prefs = Preferences.userNodeForPackage(InputBox.this.getClass)
```

Comment:
`InputBox.this.getClass` is correct, but it would be more idiomatic scala to write `classOf[InputBox]` since the class is known at compile-time. `classOf[X]` is the scala parallel to `X.class` in java.